lib/assert.js

```diff
@@ -388,22 +539,14 @@ function compareBranch(
 ) {
   // Check for Map object equality
   if (isMap(actual) && isMap(expected)) {
-    if (actual.size !== expected.size) {
-      return false;
-    }
-    const safeIterator = FunctionPrototypeCall(SafeMap.prototype[SymbolIterator], actual);
-
-    comparedObjects ??= new SafeWeakSet();
+    return compareMaps(actual, expected, comparedObjects);
+  }
 
-    for (const { 0: key, 1: val } of safeIterator) {
-      if (!MapPrototypeHas(expected, key)) {
-        return false;
-      }
-      if (!compareBranch(val, MapPrototypeGet(expected, key), comparedObjects)) {
-        return false;
-      }
-    }
-    return true;
+  if (
+    (ArrayBufferIsView(actual) && ArrayBufferIsView(expected)) ||
+    (isArrayBuffer(actual) && isArrayBuffer(expected))
```

Comment:
Don't we want `||` here to catch all use of binary data?
```suggestion
    ArrayBufferIsView(actual) || ArrayBufferIsView(expected)) ||
    isArrayBuffer(actual) || isArrayBuffer(expected)
```
Also, we may want to include `SharedArrayBuffer`
```suggestion
    ArrayBufferIsView(actual) || ArrayBufferIsView(expected)) ||
    isAnyArrayBuffer(actual) || isAnyArrayBuffer(expected)
```